### PR TITLE
Fluent interface for launcher

### PR DESCRIPTION
**Pull request by pJunger**
Replaces previous launcher functions through fluent interface. 
Added possibility to add cucumber arguments programatically.

### CODE REVIEW

**Comment by acmcarther** [comment]
I haven't taken a look yet, but minor heads up. I've been merge-squashing in this project, meaning you'll get a funky conflict when you try to sync up since you've included the changes from #36 in this PR. I can do a normal merge on #36 if you want, but it'll mess up the clean commit history so far. I'm not super opinionated either way though.

Once more, big thanks for these PRs. This one especially is a big win for ergonomics.

**Comment by acmcarther** [comment]
Addendum: Actually, I think you've _already got_ a funky merge conflict from the last time i did a merge squash.

Sorry!

**Comment by pJunger** [comment]
You're welcome! (Well, and I really did want those cmd line arguments anyway ^^).
And I really won't get around to doing anything more until I get back, sorry!

**Comment by pJunger** [comment]
Hm, must have broken something ...

**Comment by acmcarther** [comment]
taking a look

**Comment by acmcarther** [comment]
I just updated the calculator example locally (dont forget to do that!) and ran it as specified and it worked. I also made a test fail, and a test pending, and those appeared to function as they should as well.

I'm inclined to believe the issue here has something to do with having two cucumber servers running simultaneously -- specifically the server for cucumber itself, and the server for the test application.

My first thought was that it was a port issue, but I don't see anything obvious going on there. I'll keep looking.

File: features/support/fs.rs [hunk 1]
```diff
@@ -146,15 +146,15 @@ fn create_features(dir: TempDir) -> io::Result<TempDir> {
 
         fn main() {
           let world: u32 = 0;
-          cucumber::start_with_addr(\"0.0.0.0:7879\", world, &[&steps::register_steps]);
+          cucumber::create_config(world).address(\"127.0.0.1:7879\").registrar_fn(&steps::register_steps);
```
Comment (acmcarther) [inline features/support/fs.rs]: this needs to call start.

Comment (pJunger) [reply features/support/fs.rs]: ... right, thanks!

**Comment by pJunger** [comment]
Added the new calls to both examples and also fixed the call in the string template. Test output looks good :) Thank you for looking into this!

File: src/launcher.rs [hunk 2]
```diff
@@ -93,5 +136,6 @@ pub fn ruby_command() -> Command {
   env::args().skip(1).foreach(|a| {
     command.arg(a);
   });
+  command.args(args.as_slice());
```
Comment (acmcarther) [inline src/launcher.rs:141]: This addition is a bit confusing in the context of the undocumented env::args pulling behavior of this method. I'm thinking we have two sensible options:
1. Require consumer to provide env args on their own
2. Document the behavior of this method more thoroughly

I'm leaning toward (2), so heres what I might add to the documents

```
/// Build a command to execute the Ruby Cucumber Server
/// Takes a list of extra command line arguments to the server
/// NOTE: This command also passes the command line arguments passed to this executable.
```

Leaving the exact choice and wording to you.

File: .gitignore [hunk 3]
```diff
@@ -1,2 +1,4 @@
 target
 Cargo.lock
+.idea
+cucumber.iml
```
Comment (acmcarther) [inline .gitignore]: In a past life, I was told that the project gitignore should only contain things specific to the project to be ignored, and that developers should put their editor stuff in their personal global gitignore.

I actually dont really mind either way though, no change here.

**Comment by acmcarther** [comment]
This LGTM. Will merge on comment to the docs comment or change + green build.

Thanks once again!

File: .gitignore (hunk 3 above)
Comment (pJunger) [reply .gitignore]: Sounds reasonable though - will remove it again.

**Pull request merged**